common/moduleconfig/moduleconfig.go

```diff
@@ -14,12 +15,21 @@ type ModuleGoConfig struct{}
 // ModuleKotlinConfig is language-specific configuration for Kotlin modules.
 type ModuleKotlinConfig struct{}
 
+// ModuleType represents the type of the module.
+type ModuleType int
+
+const (
+	FTLModule ModuleType = iota
+	ExternalLibrary
```

Comment:
What do we think of "external library" as what we call these things?
Trying to use a name separate from ExternalModule since that's already used for imported modules